### PR TITLE
Hardware Integration

### PR DESCRIPTION
**Pull request by JamesSmartCell**
- Refactor of Transaction signing to allow hardware signing
- Implement hardware stub library to enable adding new hardware support

### CODE REVIEW

File: app/src/main/java/com/alphawallet/app/ui/widget/adapter/WalletsSummaryAdapter.java [hunk 1]
```diff
@@ -261,11 +262,30 @@ public void setWallets(Wallet[] wallets)
...
+                        w.isSynced = false;
+                        break;
                     default:
                         break;
                 }
             }
 
+            if (hasHardwareWallet)
+            {
+                Wallet hardwareText = new Wallet(context.getString(R.string.hardware_wallets));
```
Comment (seabornlee) [inline app/src/main/java/com/alphawallet/app/ui/widget/adapter/WalletsSummaryAdapter.java:276]: It's not a Text.

**Review by seabornlee — APPROVED** [review]
This is an outstanding accomplishment.

File: hardware_stub/src/main/java/com/alphawallet/hardware/HardwareCallback.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.alphawallet.hardware;
+
+/**
+ * Created by JB on 1/02/2023.
+ */
+public interface HardwareCallback
```
[the remark this answers is not in the document]
Comment (JamesSmartCell) [reply hardware_stub/src/main/java/com/alphawallet/hardware/HardwareCallback.java:6]: can you suggest a different name that makes more sense? I wasn't really happy with that name but figured a developer could see the interface was related to hardware integration.

**Comment by hboon** [comment]
@JamesSmartCell not rebase?

**Comment by JamesSmartCell** [comment]
I used rebase

**Comment by JamesSmartCell** [comment]
Don't know why it says 'merged'.

**Pull request closed without merging**